Fix Wrong ResponseCode issue, remove duplicated log {msg}, add Respon…

hujunhaorobert opened the pull request:
I had reported a bug for newman-reporter-teamcity recently. Error description:  https://github.com/leafle/newman-reporter-teamcity/issues/13 
After some investigation, I found a fix solution for index.js.  That is:
Change the code line from 
const responseCode = (this.currItem.response && this.currItem.response.responseTime) || "-";
to 
const responseCode = this.currItem.response.code;
By the way, I found some redundant log info could be removed (e.g. message='${msg}'). 
console.log(`##teamcity[testFailed name='${this.escape(args.item.name)}' message='${msg}' details='${msg} - ${details}']`);
In the meantime, as an improvement proposal, I suggest to add repsoneBody into the teamcity log when there is assertion failure, so that we can easily check the ResponseBody.

Commit:
Fix Wrong ResponseCode issue, remove duplicated log {msg}, add ResponseBody

leafle commented on lib/TeamcityReporter.js:
```diff
@@ -33,10 +33,19 @@ class TeamcityReporter {
         if (!this.currItem.passed) {
             const msg = this.escape(this.currItem.failedAssertions.join(", "));
-            const responseCode = (this.currItem.response && this.currItem.response.responseTime) || "-";
+            const responseCode = this.currItem.response.code;
```
Since `this.currItem.response` is potentially `undefined` the full check is still required.

hujunhaorobert replied:
Good comments, I will update it to 
const responseCode = (this.currItem.response && this.currItem.response.code) || "-";

leafle commented on lib/TeamcityReporter.js:
```diff
@@ -33,10 +33,19 @@ class TeamcityReporter {
             const reason = (this.currItem.response && this.currItem.response.reason()) || "-";
             const details = this.escape(`Response code: ${responseCode}, reason: ${reason}`);
-            console.log(`##teamcity[testFailed name='${this.currItem.name}' message='${msg}' details='${msg} - ${details}']`);
+            console.log(`##teamcity[testFailed name='${this.currItem.name}' details='${msg} - ${details}']`);
```
`message` is intended as a succinct description of the failure and `details` is a more verbose explanation. Neither is redundant.

hujunhaorobert replied:
I disagree on this. Reason as below:
Below is the teamcity log example in Powershell if keep it as it was without the fix solution,   
![image](https://user-images.githubusercontent.com/10079887/34802955-905e14a8-f6c4-11e7-8fb5-6bfbb32d85c6.png)
You can see the in the log , the log info "message='3) Check SemaphoreId is NOT Created for |'No Semaphore Name|' (Negative Scenario)'" has been already included in the details part obviously, which makes the log more tedious.

If take the fix solution, the team city log will remove the redundant {msg} log info, which make it better.
![image](https://user-images.githubusercontent.com/10079887/34803066-49d69554-f6c5-11e7-9b95-d70cd8a01472.png)

leafle commented on lib/TeamcityReporter.js:
```diff
@@ -33,10 +33,19 @@ class TeamcityReporter {
+			}
+			catch (e) {
+				console.log(`##teamcity[testFailed Exception : '${e}']`);
+			}
```
Putting the `responseBody` in the Teamcity message seems noisy and undesirable to me and I don't think the catch  block is logging a valid Teamcity message.

hujunhaorobert replied:
Well, fair enough. You can ignore this change proposal.

leafle reviewed (changes requested):
Thanks for reporting the issue and submitting a fix. I think we should keep it simple though and just replace `responseTime` with `responseCode` on the problematic line.

hujunhaorobert committed:
Merge branch 'master' into fixResponseCode

hujunhaorobert committed:
Enhance the responseCode

Consider the response undefined case, update the responseCode

leafle commented on lib/TeamcityReporter.js:
```diff
@@ -34,10 +34,10 @@ class TeamcityReporter {
             const reason = (this.currItem.response && this.currItem.response.reason()) || "-";
             const details = this.escape(`Response code: ${responseCode}, reason: ${reason}`);
-            console.log(`##teamcity[testFailed name='${this.currItem.name}' message='${msg}' details='${msg} - ${details}' flowId='${this.flowId}']`);
+            console.log(`##teamcity[testFailed name='${this.currItem.name}' details='${msg} - ${details}' flowId='${this.flowId}']`);
```
Teamcity uses the `message` in some cases without the `details`, for instance, in a list with a succinct description. For that reason we shouldn't remove the message attribute. You might have a point about removing `${msg}` from `details` unless there is view in the TC UI where `details` are displayed without `message`.

This change is independent of the fix you are making above though, so let's put it in a different commit so I can accept the fix.

hujunhaorobert replied:
That is fair enough, thanks for your comments. My purpose is just to remove the duplicated {msg} log info, so removing the {msg} from "details='${msg} - ${details}'" could be a better solution considering your concern. For now, I just change the responseCode in the new commit.

hujunhaorobert committed:
Skip removing the message='${msg}'

Only change the the responseCode to fix the wrong responseCode issue. Will create a separate new commit to remove the redundant {msg} log issue.

hujunhaorobert commented:
Hey @leafle , Can you review the commit in my pull request? Also can you please estimate when there will be a new release for newman-reporter-teamcity? Nowadays, our DEV team is actively checking the teamcity log, especially the failure API testing response Code. We had a strong request to have a new release. Thanks!
